### PR TITLE
Cilium BGPv1 Reconciler - Handle updated and deprecated Cidr fields for CiliumLoadBalancerIPPool

### LINKED ISSUE #32693: BGP Advertised Path Attributes fails to match CiliumLoadBalancerIPPool
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

This bug report is a follow-up from https://cilium.slack.com/archives/C53TG4J4R/p1716491937260819

I attempted to configure the Advertised Paths feature using https://docs.cilium.io/en/stable/network/bgp-control-plane/#advertised-path-attributes.  Before configuring the Advertised Paths feature, I had established BGP peering and a BGP path was being announced.  This was seen in:

```
 cilium bgp routes
(Defaulting to `available ipv4 unicast` routes, please see help for more options)

Node                              VRouter   Prefix          NextHop    Age      Attrs
bgp-cplane-dev-v4-control-plane   65001     100.64.8.2/32   10.0.1.1   41m37s   [{Origin: i} {AsPath: 65000 65002} {Nexthop: 10.0.1.1}]
bgp-cplane-dev-v4-worker          65002     100.64.8.2/32   0.0.0.0    46m52s   [{Origin: i} {Nexthop: 0.0.0.0}]
```

For the Advertised Paths feature, I had configured:
```
      advertisedPathAttributes:
      - selectorType: CiliumLoadBalancerIPPool
        selector:
          matchLabels:
            environment: production
        communities:
          standard:
          - 65001:100
```
With this configuration, no BGP communities were present on the announced path.

The full configuration is below:
```
cat bgpp.yaml

apiVersion: cilium.io/v2alpha1
kind: CiliumBGPPeeringPolicy
metadata:
  name: worker
spec:
  nodeSelector:
    matchLabels:
      kubernetes.io/hostname: bgp-cplane-dev-v4-worker
  virtualRouters:
  - localASN: 65002
    serviceSelector:
      matchExpressions:
       - { key: app, operator: In, values: [hello-node, hello-node2, hello-node3] }
    serviceAdvertisements:
      - LoadBalancerIP # <-- default
      #- ClusterIP      # <-- options
      #- ExternalIP     # <-- options
    neighbors:
    - peerASN: 65000
      peerAddress: 10.0.2.1/32
      gracefulRestart:
        enabled: true
        restartTimeSeconds: 60
      advertisedPathAttributes:
      - selectorType: CiliumLoadBalancerIPPool
        selector:
          matchLabels:
            environment: vrf-mgmt
        communities:
          standard:
          - 65001:100
---
apiVersion: "cilium.io/v2alpha1"
kind: CiliumLoadBalancerIPPool
metadata:
  name: "vrf-mgmt"
  labels:
    environment: vrf-mgmt
spec:
  blocks:
  - cidr: "100.64.8.0/22"
---
```

There were no route policies observed on the Cilium nodes:
```
$  k exec cilium-vj8nj -n kube-system -- cilium-dbg bgp route-policies
VRouter   Policy Name   Type   Match Peers   Match Prefixes (Min..Max Len)   RIB Action   Path Actions
```

```
$  k exec cilium-lj9b5 -n kube-system -- cilium-dbg bgp route-policies
VRouter   Policy Name   Type   Match Peers   Match Prefixes (Min..Max Len)   RIB Action   Path Actions
```

Other related information was:
```
$  k get ippools --show-labels
NAME                DISABLED   CONFLICTING   IPS AVAILABLE   AGE   LABELS
vrf-core-services   false      False         254             64m   <none>
vrf-customers       false      False         254             64m   <none>
vrf-mgmt            false      False         1020            64m   environment=vrf-mgmt
```

```
$  k get services --show-labels
NAME           TYPE           CLUSTER-IP     EXTERNAL-IP   PORT(S)          AGE   LABELS
hello-node     LoadBalancer   10.2.157.150   100.64.8.2    8080:32293/TCP   67m   app=hello-node
hello-node-2   LoadBalancer   10.2.88.35     100.64.8.1    8080:32414/TCP   64m   app=hello-node-2
kubernetes     ClusterIP      10.2.0.1       <none>        443/TCP          71m   component=apiserver,provider=kubernetes
```

In the Slack thread, Harsimran Pabla pointed out that the existing BGP reconciler references a deprecated CRD field (`cidrs`) for `CiliumLoadBalancerIPPool`.  In https://github.com/cilium/cilium/blob/main/pkg/bgpv1/manager/reconciler/route_policy.go, he pointed out:

```
            for _, cidrBlock := range pool.Spec.Cidrs {
```

As a test, for the `CiliumLoadBalancerIPPool ` I changed:
```
spec:
  blocks:
  - cidr: "100.64.8.0/22"
```

to:
```
spec:
  cidrs:
  - "100.64.8.0/22"
```

This corrected my issue!  The configured BGP community, via the Advertised Paths Feature, was now present on the announcement:

```
$  cilium bgp routes
(Defaulting to `available ipv4 unicast` routes, please see help for more options)

Node                              VRouter   Prefix          NextHop    Age      Attrs
bgp-cplane-dev-v4-control-plane   65001     100.64.8.2/32   10.0.1.1   7s       [{Origin: i} {AsPath: 65000 65002} {Nexthop: 10.0.1.1} {Communities: 65001:100}]
bgp-cplane-dev-v4-worker          65002     100.64.8.2/32   0.0.0.0    59m12s   [{Origin: i} {Nexthop: 0.0.0.0}]
```

```
$  k exec cilium-vj8nj -n kube-system -- cilium-dbg bgp route-policies
VRouter   Policy Name                                   Type     Match Peers   Match Prefixes (Min..Max Len)   RIB Action   Path Actions
65002     10.0.2.1/32-CiliumLoadBalancerIPPool-52e...   export   10.0.2.1/32   100.64.8.0/22 (32..32)          none         AddCommunities: [65001:100]
```

```
$  k exec cilium-vj8nj -n kube-system -- cilium-dbg bgp route-policies
VRouter   Policy Name                                   Type     Match Peers   Match Prefixes (Min..Max Len)   RIB Action   Path Actions
65002     10.0.2.1/32-CiliumLoadBalancerIPPool-52e...   export   10.0.2.1/32   100.64.8.0/22 (32..32)          none         AddCommunities: [65001:100]
```

This bug request is being opened to capture the necessary changes to https://github.com/cilium/cilium/blob/main/pkg/bgpv1/manager/reconciler/route_policy.go to allow one to specify either the deprecated (`cidrs`) or current field (`blocks`).

### Cilium Version

```
$  cilium version
cilium-cli: v0.16.7-40-g9316d0ac compiled with go1.22.2 on linux/amd64
cilium image (default): v1.15.5
cilium image (stable): v1.15.5
cilium image (running): 1.16.0-dev
```

### Kernel Version

```
$  uname -a
Linux hostname 6.5.0-26-generic #26~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Tue Mar 12 10:22:43 UTC 2 x86_64 x86_64 x86_64 GNU/Linux
```

### Kubernetes Version

```
$  k version
Client Version: version.Info{Major:"1", Minor:"20+", GitVersion:"v1.20.15-enhanced-describe-dirty", GitCommit:"ac2e2baa7d4039cc4c68f2e869e4edbe2d60b305", GitTreeState:"dirty", BuildDate:"2023-03-02T00:33:46Z", GoVersion:"go1.20.1", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"30", GitVersion:"v1.30.0", GitCommit:"7c48c2bd72b9bf5c44d21d7338cc7bea77d0ad2a", GitTreeState:"clean", BuildDate:"2024-05-13T22:00:36Z", GoVersion:"go1.22.2", Compiler:"gc", Platform:"linux/amd64"}
```

### Regression

_No response_

### Sysdump

[cilium-sysdump-20240523-133822.zip](https://github.com/cilium/cilium/files/15422407/cilium-sysdump-20240523-133822.zip)

### Relevant log output

_No response_

### Anything else?

_No response_

### Cilium Users Document

- [X] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by dswaffordcw**
## Cilium BGPv1 Reconciler - Handle updated and deprecated Cidr fields for CiliumLoadBalancerIPPool

Please ensure your pull request adheres to the following guidelines:

- [x] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [x] All code is covered by unit and/or runtime tests where feasible.
- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [x] If your commit description contains a `Fixes: <commit-id>` tag, then
      please add the commit author[s] as reviewer[s] to this issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [x] Provide a title or release-note blurb suitable for the release notes.
- [x] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)
- [x] Thanks for contributing!

Fixes: #32693

## Description
In https://github.com/cilium/cilium/commit/27322f3959c3fa05b9b1c4f9827527b4a3642687, the CiliumLoadBalancerIPPool's field named "cidrs" was deprecated.  The documentation on https://docs.cilium.io/en/stable/network/lb-ipam/ provides an example of configuring a CiliumLoadBalancerIPPool using the field named "blocks".  While testing a BGP policy configured with the Advertised Path Attributes feature (https://docs.cilium.io/en/stable/network/bgp-control-plane/#advertised-path-attributes), I was not able to achieve the desired policy.  BGP attributes configured were not being applied.

While discussing this in Cilium's Slack channel, it was pointed out that the BGPv1 reconciler was only aware of the deprecated field.

This commit updates Cilium's BGPv1 reconciler to support both the deprecated and updated fields.

## Unit Tests
Unit tests were updated as follows:
* To ensure that the deprecated field continues to be supported
* To ensure that the new field is supported
* To ensure that mixing both new and deprecated fields is supported
* To ensure that mixing both new and deprecated fields containing the same prefix is handled (by removing duplicate prefixes)

## Manual Testing
For manual testing, I brought up the BGP development environment (see https://docs.cilium.io/en/stable/contributing/development/bgp_cplane/).

Steps included:
```
make kind-bgp-v4

KIND_CLUSTER_NAME=bgp-cplane-dev-v4 make kind-image
cilium install --chart-directory install/kubernetes/cilium -f contrib/containerlab/bgp-cplane-dev-v4/values.yaml --set image.override="localhost:5000/cilium/cilium-dev:local" --set image.pullPolicy=Never --set operator.image.override="localhost:5000/cilium/operator-generic:local" --set operator.image.pullPolicy=Never
```

Next, I applied the policy below:
```
cat bgpp.yaml
---
apiVersion: cilium.io/v2alpha1
kind: CiliumBGPPeeringPolicy
metadata:
  name: control-plane
spec:
  nodeSelector:
    matchLabels:
      kubernetes.io/hostname: bgp-cplane-dev-v4-control-plane
  virtualRouters:
  - localASN: 65001
    neighbors:
    - peerASN: 65000
      peerAddress: 10.0.1.1/32
---
apiVersion: cilium.io/v2alpha1
kind: CiliumBGPPeeringPolicy
metadata:
  name: worker
spec:
  nodeSelector:
    matchLabels:
      kubernetes.io/hostname: bgp-cplane-dev-v4-worker
  virtualRouters:
  - localASN: 65002
    serviceSelector:
      matchExpressions:
       - { key: app, operator: In, values: [hello-node, hello-node2, hello-node3] }
    serviceAdvertisements:
      - LoadBalancerIP # <-- default
      #- ClusterIP      # <-- options
      #- ExternalIP     # <-- options
    neighbors:
    - peerASN: 65000
      peerAddress: 10.0.2.1/32
      gracefulRestart:
        enabled: true
        restartTimeSeconds: 60
      advertisedPathAttributes:
      - selectorType: CiliumLoadBalancerIPPool
        selector:
          matchLabels:
            environment: mgmt
        communities:
          standard:
          - 65002:100
          large:
          - 65002:100:1
---
apiVersion: "cilium.io/v2alpha1"
kind: CiliumLoadBalancerIPPool
metadata:
  name: "mgmt"
  labels:
    environment: mgmt
spec:
  cidrs:
    - cidr: "100.64.8.0/22"
```

Using:
```
k apply -f bgpp.yaml
```

Next, I checked the state of the peers:
```
$  cilium bgp peers
Node                              Local AS   Peer AS   Peer Address   Session State   Uptime   Family         Received   Advertised
bgp-cplane-dev-v4-control-plane   65001      65000     10.0.1.1       established     3m24s    ipv4/unicast   0          0
                                                                                               ipv6/unicast   0          0
bgp-cplane-dev-v4-worker          65002      65000     10.0.2.1       established     3m27s    ipv4/unicast   0          1
                                                                                               ipv6/unicast   0          1
```

At this point, I have not yet spun up any services/pods.  I then brought up a "hello-world" application using:

```
kubectl create deployment hello-node --image=registry.k8s.io/e2e-test-images/agnhost:2.39 -- /agnhost netexec --http-port=8080

kubectl expose deployment hello-node --type=LoadBalancer --port=8080
```

And observed:
```
$  k get services
NAME         TYPE           CLUSTER-IP    EXTERNAL-IP   PORT(S)          AGE
hello-node   LoadBalancer   10.2.118.17   100.64.8.1    8080:30934/TCP   21s
```

Followed by:
```
$  cilium bgp routes
(Defaulting to `available ipv4 unicast` routes, please see help for more options)

Node                              VRouter   Prefix          NextHop    Age   Attrs
bgp-cplane-dev-v4-control-plane   65001     100.64.8.1/32   10.0.1.1   38s   [{Origin: i} {AsPath: 65000 65002} {Nexthop: 10.0.1.1} {Communities: 65002:100} {LargeCommunity: [ 65002:100:1]}]
bgp-cplane-dev-v4-worker          65002     100.64.8.1/32   0.0.0.0    38s   [{Origin: i} {Nexthop: 0.0.0.0}]
```

Prior to this PR, when a prefix was defined on a `CiliumLoadBalancerIPPool` using `cidrs`, there were no routes advertised.  In the above, we see routes advertised.

Next, I tore down the lab by running:
```
make kind-bgp-v4-down
```

After rebuilding the lab, I changed to: 
```
---
apiVersion: "cilium.io/v2alpha1"
kind: CiliumLoadBalancerIPPool
metadata:
  name: "mgmt"
  labels:
    environment: mgmt
spec:
  blocks:
  - cidr: "100.64.8.0/22"
```

After applying, I see routes as expected:
```
$  cilium bgp routes
(Defaulting to `available ipv4 unicast` routes, please see help for more options)

Node                              VRouter   Prefix          NextHop    Age     Attrs
bgp-cplane-dev-v4-control-plane   65001     100.64.8.1/32   10.0.1.1   1m34s   [{Origin: i} {AsPath: 65000 65002} {Nexthop: 10.0.1.1} {Communities: 65002:100} {LargeCommunity: [ 65002:100:1]}]
bgp-cplane-dev-v4-worker          65002     100.64.8.1/32   0.0.0.0    1m42s   [{Origin: i} {Nexthop: 0.0.0.0}]
```

In one test, I specified both the new and deprecated fields:
```
apiVersion: "cilium.io/v2alpha1"
kind: CiliumLoadBalancerIPPool
metadata:
  name: "mgmt"
  labels:
    environment: mgmt
spec:
  blocks:
    - cidr: "100.64.8.0/22"
  cidrs:
    - cidr: "100.64.12.0/22"
```

Here, I observed the last prefix being used:
```
$  k get services
NAME         TYPE           CLUSTER-IP     EXTERNAL-IP   PORT(S)          AGE
hello-node   LoadBalancer   10.2.238.103   100.64.12.1   8080:30608/TCP   2m30s
kubernetes   ClusterIP      10.2.0.1       <none>        443/TCP          4m34s

$  cilium bgp routes
(Defaulting to `available ipv4 unicast` routes, please see help for more options)

Node                              VRouter   Prefix           NextHop    Age     Attrs
bgp-cplane-dev-v4-control-plane   65001     100.64.12.1/32   10.0.1.1   1m36s   [{Origin: i} {AsPath: 65000 65002} {Nexthop: 10.0.1.1} {Communities: 65002:100} {LargeCommunity: [ 65002:100:1]}]
bgp-cplane-dev-v4-worker          65002     100.64.12.1/32   0.0.0.0    1m44s   [{Origin: i} {Nexthop: 0.0.0.0}]
```

```release-note
Cilium BGPv1 Reconciler - Handle updated and deprecated Cidr fields for CiliumLoadBalancerIPPool
```

### CODE REVIEW

**Comment by rastislavs** [comment]
@dswaffordcw Thanks for the fix! Looks good, could you please also extend the test in `route_policy_test.go` to cover this case as well?

**Comment by rastislavs** [comment]
Note: the `CiliumLoadBalancerIPPool.Spec.Cidrs` field has been deprecated as of https://github.com/cilium/cilium/commit/27322f3959c3fa05b9b1c4f9827527b4a3642687, so we also need a backport to v1.15 (labeling for backport).

**Comment by harsimran-pabla** [comment]
Hi @dswaffordcw Thanks for the quick fix. Couple of suggestions
- Add sign-off in the commit message.
- Add a brief commit message explaining the change.

**Comment by dswaffordcw** [comment]
> @dswaffordcw Thanks for the fix! Looks good, could you please also extend the test in `route_policy_test.go` to cover this case as well?

Hi @rastislavs ,

I updated the tests as follows
* The existing test was changed to reference the field `blocks`
* A copy of the existing test was made and continues to reference the original field `cidrs`. 
* Another copy of the existing test was made that specifies different prefixes using `blocks` and `cidrs`.
* Another copy of the existing test was made that specifies the same prefix using both `blocks` and `cidrs`.  This test exercises a check that prevents duplicates.  When the same prefix is specified in both new and old fields, the duplicate is ignored.

I don't like all the copy-and-paste that's going on there.  I thought about re-factoring some of that test out with another function or two but held off.  Let me know if you'd like me to explore that.

File: pkg/bgpv1/manager/reconciler/route_policy.go [hunk 1]
```diff
@@ -238,8 +238,28 @@ func (r *RoutePolicyReconciler) pathAttributesToPolicy(attrs v2alpha1api.CiliumB
 			if attrs.Selector != nil && !labelSelector.Matches(labels.Set(pool.Labels)) {
 				continue
 			}
+			prefixesSeen := map[netip.Prefix]bool{}
```
Comment (harsimran-pabla) [inline pkg/bgpv1/manager/reconciler/route_policy.go]: nit: In most of BGP package, sets utility is used for this sort of operation. Consider using [sets](https://pkg.go.dev/k8s.io/apimachinery/pkg/util/sets) package ( `prefixesSeen := sets.New[netip.Prefix]()` )

File: pkg/bgpv1/manager/reconciler/route_policy.go [hunk 2]
```diff
@@ -238,8 +238,28 @@ func (r *RoutePolicyReconciler) pathAttributesToPolicy(attrs v2alpha1api.CiliumB
...
+			for _, cidrBlock := range pool.Spec.Blocks {
+				cidr, err := netip.ParsePrefix(string(cidrBlock.Cidr))
+				prefixesSeen[cidr] = true
```
Comment (harsimran-pabla) [inline pkg/bgpv1/manager/reconciler/route_policy.go]: nit/minor stying: I'd prefer to set `prefixesSeen` after evaluating the error. It is not an issue in current usage, but it has potential to become one if we add more logic around `prefixesSeen`.

File: pkg/bgpv1/manager/reconciler/route_policy.go [hunk 3]
```diff
@@ -238,8 +238,28 @@ func (r *RoutePolicyReconciler) pathAttributesToPolicy(attrs v2alpha1api.CiliumB
...
+				} else {
+					v6Prefixes = append(v6Prefixes, &types.RoutePolicyPrefixMatch{CIDR: cidr, PrefixLenMin: maxPrefixLenIPv6, PrefixLenMax: maxPrefixLenIPv6})
+				}
+			}
+			// Note: CiliumLoadBalancerIPPool.Spec.Cidrs was deprecated as of
+			// https://github.com/cilium/cilium/commit/27322f3959c3fa05b9b1c4f9827527b4a3642687
+			// It was replaced by CiliumLoadBalancerIPPool.Spec.Blocks.
 			for _, cidrBlock := range pool.Spec.Cidrs {
 				cidr, err := netip.ParsePrefix(string(cidrBlock.Cidr))
+				// If the same prefix was specified in Spec.Blocks and Spec.Cidrs, ignore the duplicate.
```
Comment (harsimran-pabla) [inline pkg/bgpv1/manager/reconciler/route_policy.go]: Same here, to move this check after evaluating error.

**Comment by harsimran-pabla** [comment]
Thanks @dswaffordcw for the changes. Some minor comments, but overall change looks good.

> I don't like all the copy-and-paste that's going on there. I thought about re-factoring some of that test out with another function or two but held off. Let me know if you'd like me to explore that.

Let's hold on to that for now - this code will eventually deprecate as we move to BGPv2 reconcilers. Please let me know if you want to contribute to the BGP subsystem. There is plenty to do :)

**Comment by harsimran-pabla** [comment]
One natural change is to make sure BGPv2 Reconciler also have similar fix. Currently, it is only checking for Blocks [here](https://github.com/cilium/cilium/blob/main/pkg/bgpv1/manager/reconcilerv2/service.go#L804).

File: pkg/bgpv1/manager/reconciler/route_policy.go (hunk 1 above)
Comment (dswaffordcw) [reply pkg/bgpv1/manager/reconciler/route_policy.go]: That's a great nit!  I hadn't run across that package before.

File: pkg/bgpv1/manager/reconciler/route_policy.go (hunk 2 above)
Comment (dswaffordcw) [reply pkg/bgpv1/manager/reconciler/route_policy.go]: Good eye.  Updated.

File: pkg/bgpv1/manager/reconciler/route_policy.go (hunk 3 above)
Comment (dswaffordcw) [reply pkg/bgpv1/manager/reconciler/route_policy.go]: Updated.

**Review by rastislavs — APPROVED** [review]
LGTM, thanks a lot for updating the tests as well!

**Review by harsimran-pabla — APPROVED** [review]
🚀

**Comment by dswaffordcw** [comment]
@harsimran-pabla Any thoughts on what is failing with "BPF Checks / Check Patch"?

I ran that locally and it seemed to work:

```
> cilium-fork on  bug/GH_ISSUE_32693 [$]
$  make -C bpf checkpatch
make: Entering directory '/home/dswafford/code/cilium-fork/bpf'
docker container run --rm \
        --workdir /workspace \
        --volume /home/dswafford/code/cilium-fork/bpf/..:/workspace \
        --user "1000:1000" \
        -e GITHUB_REF= -e GITHUB_REPOSITORY= \
        --entrypoint /bin/bash quay.io/cilium/cilium-checkpatch:40ae57b72ea430ab77b45fbed9a1cd4295838ef6@sha256:ccd20b54c779c4645e72dbb48bfcdacb9929cae5d334b0fa6d1ef996af64348f /checkpatch/checkpatch.sh -- --ignore MACRO_ARG_REUSE
Retrieved 1 commits on top of ref upstream/main

=========================================================
[1/1] Running on 91c528be44c414dcadd10367fb9038075725180a
Cilium BGPv1 Reconciler - Handle updated and deprecated Cidr fields
=========================================================

"[PATCH] Cilium BGPv1 Reconciler - Handle updated and deprecated Cidr" has no obvious style problems and is ready for submission.

NOTE: Ignored message types: BIT_MACRO C99_COMMENT_TOLERANCE COMMIT_LOG_LONG_LINE COMMIT_MESSAGE COMPLEX_MACRO CONSTANT_CONVERSION CONST_STRUCT EMAIL_SUBJECT FILE_PATH_CHANGES FROM_SIGN_OFF_MISMATCH GIT_COMMIT_ID JIFFIES_COMPARISON LEADING_SPACE LONG_LINE_COMMENT MACRO_ARG_REUSE MACRO_WITH_FLOW_CONTROL MULTISTATEMENT_MACRO_USE_DO_WHILE NOT_UNIFIED_DIFF PRINTK_WITHOUT_KERN_LEVEL TRAILING_SEMICOLON TRAILING_STATEMENTS VOLATILE
All done
make: Leaving directory '/home/dswafford/code/cilium-fork/bpf'

00x0 dswafford@dswafford-2 2024/05/29 11:48:16
☸︎ teleport.rno2.int.coreweave.com:
> cilium-fork on  bug/GH_ISSUE_32693 [$]
$  echo $?
0
```
```

**Comment by harsimran-pabla** [comment]
Hi @dswaffordcw I think check is still failing because of merge commit 5d8961c8, instead of merging the main branch into this branch - rebasing this branch on top of the main should resolve this issue.

**Comment by dswaffordcw** [comment]
Thanks @harsimran-pabla!  You were right.  I had messed up the rebase w/ my local fork.

**Pull request merged**